test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +91,10 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        Command command = new Parser().parseCommand(inputCommand);
+        if (command.isMutating()) {
+            assertEquals(addressBook, saveFile.load());
+        }
```

Comment:
This change doesn’t help much because it does not confirm that the saving happens _only_ for mutating commands. Tests will pass even if you don’t have this change. To be complete, you should have an `else` branch that confirms the file was not updated for non mutating commands (e.g. by checking the last updated time of the file).